File: easy/0896. Monotonic Array/main.java
// Solved by @kitanoyoru
// https://leetcode.com/problems/monotonic-array/submissions/

class Solution {
  public boolean isMonotonic(int[] nums) {
    int incCount = 0, decrCount = 0;

    for (int i = 1; i < nums.length; i++) {
      if (nums[i] < nums[i-1]) {
        incCount++;
      } else if (nums[i] > nums[i-1]) {
        decrCount++;
      }
    }

    return !((incCount != 0 ? true : false) && (decrCount != 0 ? true : false));
  }
}

File: easy/0896. Monotonic Array/main.go
package main

func isMonotonic(nums []int) bool {
	inc, dec := false, false

	for i := 1; i < len(nums); i++ {
		if nums[i] < nums[i-1] {
			inc = true
		} else if nums[i] > nums[i-1] {
			dec = true
		}
	}

	return !(inc && dec)
}
